### PR TITLE
Dagger 2 minimum

### PR DESCRIPTION
**Pull request by pollend**
# Description
this is a minimum implementation of dagger 2 for the application screen. I need to pair down this commit more. going forward most of the getter and setters in SolApplication can be removed since the DI will manage those relationships. the game will be divided into different layers for the associated life-cycle. this should manage the lifetime of objects in the engine and also manage the injection of dependencies within the engine.

I want to get rid of context and remove assets from the global scope. will simply inject assets into the associate layer from the top of the scope down. I think this would make the code more portable.

image

### CODE REVIEW

**Comment by GooeyHub** [comment]
Can one of the admins please verify this patch?

File: desktop/src/main/java/org/destinationsol/desktop/SolDesktop.java [hunk 1]
```diff
@@ -82,6 +82,7 @@ public static void main(String[] argv) {
         handleCrashReporting(argv);
 
         // Everything is set up correctly, launch the application
+
```
Comment (Adrijaned) [inline desktop/src/main/java/org/destinationsol/desktop/SolDesktop.java]: Please remove whitespace

File: engine/build.gradle [hunk 2]
```diff
@@ -1,7 +1,12 @@
+plugins {
+    id "net.ltgt.apt" version "0.10"
+}
+
 apply from: '../config/gradle/common.gradle'
 
 [compileJava, compileTestJava]*.options*.encoding = 'UTF-8'
 
+
```
Comment (Adrijaned) [inline engine/build.gradle]: ^

File: engine/src/main/java/org/destinationsol/SolApplication.java [hunk 3]
```diff
@@ -40,65 +47,74 @@
...
-    private OggSoundManager soundManager;
-    private SolInputManager inputManager;
-
-    private UiDrawer uiDrawer;
     private MenuScreens menuScreens;
-    private SolLayouts layouts;
-    private GameOptions options;
-    private CommonDrawer commonDrawer;
+
+
```
Comment (Adrijaned) [inline engine/src/main/java/org/destinationsol/SolApplication.java]: ^

File: engine/src/main/java/org/destinationsol/SolApplication.java [hunk 4]
```diff
@@ -40,65 +47,74 @@
...
 
     @Override
     public void create() {
-        context = new ContextImpl();
-        context.put(SolApplication.class, this);
-        worldConfig = new WorldConfig();
-        isMobile = Gdx.app.getType() == Application.ApplicationType.Android || Gdx.app.getType() == Application.ApplicationType.iOS;
-        if (isMobile) {
+
+
```
Comment (Adrijaned) [inline engine/src/main/java/org/destinationsol/SolApplication.java]: Another unnecessary whitespace

File: engine/src/main/java/org/destinationsol/assets/Assets.java [hunk 5]
```diff
@@ -236,4 +235,4 @@ public static void uncacheLists() {
 
         return textures;
     }
-}
+}
```
Comment (Adrijaned) [inline engine/src/main/java/org/destinationsol/assets/Assets.java]: ?

File: engine/src/main/java/org/destinationsol/assets/audio/OggMusicManager.java [hunk 6]
```diff
@@ -16,9 +16,13 @@
 package org.destinationsol.assets.audio;
 
 import com.badlogic.gdx.audio.Music;
+import dagger.Provides;
 import org.destinationsol.GameOptions;
+import org.destinationsol.assets.AssetHelper;
 import org.destinationsol.assets.Assets;
 
+import javax.inject.Inject;
+import javax.inject.Singleton;
```
Comment (Adrijaned) [inline engine/src/main/java/org/destinationsol/assets/audio/OggMusicManager.java]: This seems like a bunch of unused imports

File: engine/src/main/java/org/destinationsol/game/item/ItemManager.java [hunk 7]
```diff
@@ -242,4 +242,4 @@ public void addAllGuns(ItemContainer ic) {
         }
         return res;
     }
-}
+}
```
Comment (Adrijaned) [inline engine/src/main/java/org/destinationsol/game/item/ItemManager.java]: ?

File: engine/src/main/java/org/destinationsol/common/SolRandom.java [hunk 8]
```diff
@@ -129,7 +129,7 @@ public static int randomInt(int max) {
     /**
      * Returns a random int v such that max*percentage <= v < max.
      *
-     * @param percentage The ratio of lower bound to upper bound.
+     * @param percentage The r of lower bound to upper bound.
```
Comment (Adrijaned) [inline engine/src/main/java/org/destinationsol/common/SolRandom.java]: ?

**Comment by Cervator** [comment]
Going to merge this into a target dagger branch to play with further, maybe during idle moments on my trip, easier with a formal branch for that :-)

@pollend thanks for starting on this and getting so far, I'm excited to see where we can take this!
